torch_geometric/nn/models/captum.py

```diff
@@ -237,37 +287,52 @@ def captum_output_to_dicts(
-               with key `NodeType` and value node mask tensor of shape
-               :obj:`[num_nodes, num_features]` and no edge attribution.
-
-            3. :obj:`"node_and_edge"`: :obj:`captum_attrs` contains only node
-               attributions. The returned tuple contains node attribution
-               dictionary followed by edge attribution dictionary.
+                attributions. The returned tuple has one node attribution
+                dictonary or several dictionaries if more than one object is
+                explained. The node attribution dictionary has key `NodeType`
+                and value node mask tensor of shape
+                :obj:`[num_nodes, num_features]` and no edge attribution.
+
+            3. :obj:`"node_and_edge"`: :obj:`captum_attrs` contains both node
+                and edge attributions. The returned tuple contains one node
+                attribution dictionary followed by one edge attribution
+                dictionary or several dictionaries if more than one object
+                is explained.
 
         metadata (Metadata): The metadata of the heterogeneous graph.
     """
     _raise_on_invalid_mask_type(mask_type)
     node_types = metadata[0]
     edge_types = metadata[1]
-    x_attr_dict, edge_attr_dict = None, None
-    captum_attrs = [captum_attr.squeeze(0) for captum_attr in captum_attrs]
-    if mask_type == "node":
-        assert len(node_types) == len(captum_attrs)
-        x_attr_dict = dict(zip(node_types, captum_attrs))
-    elif mask_type == "edge":
-        assert len(edge_types) == len(captum_attrs)
-        edge_attr_dict = dict(zip(edge_types, captum_attrs))
-    elif mask_type == "node_and_edge":
-        assert len(edge_types) + len(node_types) == len(captum_attrs)
-        x_attr_dict = dict(zip(node_types, captum_attrs[:len(node_types)]))
-        edge_attr_dict = dict(zip(edge_types, captum_attrs[len(node_types):]))
-    return x_attr_dict, edge_attr_dict
+    x_attr_dicts, edge_attr_dicts = [], []
+    for i in range(captum_attrs[0].shape[0]):
+        captum_attrs_i = [
+            captum_attr[i].squeeze(0) for captum_attr in captum_attrs
+        ]
+        if mask_type == "node":
+            assert len(node_types) == len(captum_attrs_i)
+            x_attr_dicts.append(dict(zip(node_types, captum_attrs_i)))
+        elif mask_type == "edge":
+            assert len(edge_types) == len(captum_attrs_i)
+            edge_attr_dicts.append(dict(zip(edge_types, captum_attrs_i)))
+        elif mask_type == "node_and_edge":
+            assert len(edge_types) + len(node_types) == len(captum_attrs_i)
+            x_attr_dicts.append(
+                dict(zip(node_types, captum_attrs_i[:len(node_types)])))
+            edge_attr_dicts.append(
+                dict(zip(edge_types, captum_attrs_i[len(node_types):])))
+    if len(x_attr_dicts) == 1:
+        x_attr_dicts = x_attr_dicts[0]
+    if len(edge_attr_dicts) == 1:
+        edge_attr_dicts = edge_attr_dicts[0]
```

Comment:
Lets return a `List` even if it of length 1. Else we'll have to update the return type hint.